### PR TITLE
Add user configurable PostCSS plugins.

### PR DESCRIPTION
**Pull request by joshgillies**
Placeholder PR to add support for configurable PostCSS Plugins.

Ref: #12

### CODE REVIEW

**Comment by joshgillies** [comment]
There's still work that needs to be done. Specifically around configuring PostCSS plugins (eg. autoprefixer). As certain PostCSS plugins are in of themselves configurable, how do we manage this?

A good example of this comes from: https://github.com/code42day/postcss-cli#examples

```
postcss --use autoprefixer --autoprefixer.browsers "> 5%" \
    --use postcss-cachify --postcss-cachify.baseUrl /res \
    -o screen.css screen.css
```

File: index.js [hunk 1]
```diff
@@ -13,6 +14,9 @@ module.exports = function (browserify, options) {
     throw new Error('css-modulesify needs the --output / -o option (path to output css file)');
   }
 
+  // PostCSS plugins passed to FileSystemLoader
+  var plugins = options.plugins || options.p || Core.defaultPlugins;
```
Comment (joshwnj) [inline index.js]: @joshgillies do you know if `plugins` works as strings or does `css-modules-loader-core` expect an array of objects?

Comment (joshgillies) [inline index.js]: Sorry, it expects an array! Good catch. :+1:

**Comment by joshgillies** [comment]
@joshwnj OK should be requiring plugins via user input now. Which I guess handles this CLI use case:

`browserify -p [ css-modulesify -p postcss-color-function -p postcss-modules-extract-imports -p postcss-modules-scope -p autoprefixer -o dist/main.css ] -o dist/index.js`

However, assuming the following use case:

```js
var browserify = require('browserify')
b.bundle('./bundle.js')
b.plugin('css-modulesify', {
  output: './bundle.css',
  plugins: [
    require('postcss-color-function'),
    require('postcss-modules-extract-imports'),
    require('postcss-modules-scope'),
    require('autoprefixer')
  ]
});
```
Maybe we should also be checking to see whether the plugin has already been `require`'d somehow...

What do you think?

**Comment by joshgillies** [comment]
The above concerns should now be addressed. An updated example would look something like this:

`browserify -p [ css-modulesify -u postcss-color-function -u postcss-modules-extract-imports -u postcss-modules-scope -u autoprefixer --autoprefixer.browsers "> 5%" -o dist/main.css ] -o dist/index.js`

Thoughts?

**Comment by joshwnj** [comment]
Looking great @joshgillies , this will be a huge improvement! Have you given it a trial run in the https://github.com/css-modules/browserify-demo yet? Will be good to demonstrate this new behaviour somewhere at any rate

**Comment by joshgillies** [comment]
I've been testing the demo over here, https://github.com/joshgillies/browserify-demo/tree/plugins. For the most part it workds, however `autoprefixer` seems to throw strange errors relating to `[TypeError: Cannot read property '_autoprefixerDisabled' of undefined]`. Given this is (at least in my mind) the defacto standard of PostCSS plugins I'd say we need to ensure css-modules play nicely with it before merging any work done here.

**Comment by joshgillies** [comment]
Ok, now that https://github.com/postcss/autoprefixer-core/issues/57 has been merged, we should be in a good position to roll this out.

I've just published some documentation around how PostCSS plugins can be used and configured.

@joshwnj would you care to take a look?

As an aside I'd also like to squash some of these commits before we merge. :smile:

**Comment by joshwnj** [comment]
@joshgillies looks great, merge when ready :)

**Pull request merged**